core/network/src/protocol.rs

```diff
@@ -743,25 +789,20 @@ impl<B: BlockT, S: NetworkSpecialization<B>, H: ExHashT> Protocol<B, S, H> {
 
 		let extrinsics = self.transaction_pool.transactions();
 		let mut propagated_to = HashMap::new();
-		for (who, ref mut peer) in self.context_data.peers.iter_mut() {
+		for (who, peer) in self.context_data.peers.iter_mut() {
 			let (hashes, to_send): (Vec<_>, Vec<_>) = extrinsics
 				.iter()
 				.filter(|&(ref hash, _)| peer.known_extrinsics.insert(hash.clone()))
 				.cloned()
 				.unzip();
 
 			if !to_send.is_empty() {
-				let (sender, port) = channel::unbounded();
-				let _ = self
-					.network_chan
-					.send(NetworkMsg::GetPeerId(who.clone(), sender));
-				let node_id = port.recv().expect("1. We are running 2. Network should be running too.");
-				if let Some(id) = node_id {
+				if let Some(id) = peer.info.peer_id.as_ref() {
 					for hash in hashes {
 						propagated_to
 							.entry(hash)
 							.or_insert_with(Vec::new)
-							.push(id.clone());
+							.push(id.to_base58());
```

Comment:
Why this change? `PeerId` implements `Hash`, so it should be possible to insert it.

Reply:
The code seems to be expecting a `String`. Note that this isn't really a change because previously the call to `to_base58` was done at https://github.com/paritytech/substrate/pull/1857/files#diff-a045635f3eec17c06b3c7ac7944acceaL522